Eastern-Research-Group/expert-query
# Feature/128 combo select

## Related Issues:
* [EQ-128](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-128)

## Main Changes:
* Migrated the source selection component from **How's My Waterway**'s [State & Tribal](https://mywaterway.epa.gov/state-and-tribal) page to a `SourceSelect` component, using **USWDS** style classes & icons instead of **ESRI**'s.
* Restructured some styles.
* Added a `context` field to the domain values ETL task, which pulls in values for an extra field (as specified in `domainValueMappings.json`).
* Added a `context` field to `fields.ts`. The `SourceSelect` component will be conditionally rendered beside a `Select` input if this field is non-empty.
* Added the ability to filter out specific values from the **ATTAINS** Domain Values service.

## Steps To Test:
1. In the `etl` directory, run `npm run etl_domain_values`.
2. Start the client application, then navigate to http://localhost:3000/attains#dataProfile=assessments.
3. Scroll down to find the field *Organization ID*.
4. Click the arrow on the right-hand side of the input, and confirm that `21FL303D` and `PUEBLO_POJOAQUE` both appear in the list.
5. Select `Tribe` from the filter list on the left-hand side of the input, and confirm that `21FL303D` no longer appears in the option list. Select `REDLAKE`.
6. Select `State` from the filter list, and confirm that `PUEBLO_POJOAQUE` no longer appears in the option list. `REDLAKE`, however, should still be listed as a selection.
7. Confirm that selected organization IDs are added to the queries displayed in the copy boxes at the bottom of the page, but not the organization types (State / Tribe).
